src/middleware.ts

```diff
@@ -325,6 +325,10 @@ export const persist = <S extends State>(
               deserializedStorageValue.version
             )
           }
+          console.error(
+            `State loaded from storage couldn't be migrated since no migrate function was provided`
+          )
+          stateFromStorageInSync = deserializedStorageValue.state
```

Comment:
```suggestion
```
we shouldn't hydrate the store with outdated data

Reply:
Thanks for catching that. 🙈